src/map/itemdb.cpp

```diff
@@ -2997,11 +2996,46 @@ uint8 ItemGroupDatabase::pc_get_itemgroup(uint16 group_id, bool identify, map_se
 	}
 	if (group->random.empty())
 		return 0;
-	
-	// Get all the 'must' item(s) (subgroup 0)
+
+	size_t count = 0;
+	// Count all the 'must' item(s) (subgroup 0)
 	uint16 subgroup = 0;
 	std::shared_ptr<s_item_group_random> random = util::umap_find(group->random, subgroup);
 
+	if (random != nullptr && !random->data.empty()) {
+		count = random->data.size();
+	}
+
+	// Count all 'random' subgroups
+	for (const auto& random : group->random) {
+		if (random.first == 0 || random.second->data.empty())
+			continue;
+		count++;
```

Comment:
Here it is hard to check, but since you only get one out of each group I would use the approach that you used for "must" and increase the counter by the amount of groups.